### PR TITLE
ROX-21046: Alerts for tenant nearing OOM

### PR DESCRIPTION
**Pull request by ludydoo**
Adding alerts for tenant containers that are about to oom

### CODE REVIEW

**Comment by stehessel** [comment]
Have you checked how close we are too hitting these alerts on prod right now? I've observed high memory usage on scanner pods in particular. See also https://redhat-internal.slack.com/archives/C0313JYKH8W/p1700683356311839?thread_ts=1700679380.899679&cid=C0313JYKH8W. I'm afraid this will be a very busy alert as is.

File: resources/prometheus/prometheus-rules.yaml [hunk 1]
```diff
@@ -159,6 +159,38 @@ spec:
             description: "Fleetshard synchronizer manages `{{ $value }}` centrals. The number of Centrals should always be larger than zero in a working system. If it drops to or below zero, fleetshard synchronizer is assumed to be in a failed state."
             sop_url: "https://gitlab.cee.redhat.com/stackrox/acs-managed-service-runbooks/blob/master/sops/dp-007-fleetshard-sync-reconciliation-error.md"
 
+    - name: tenant-resources
+      rules:
+        - expr: |
+            sum(container_memory_max_usage_bytes{namespace=~"rhacs-.{20}",container!="POD",container!=""}) by (namespace, container, pod)
```
Comment (stehessel) [inline resources/prometheus/prometheus-rules.yaml:165]:
```suggestion
            sum(container_memory_working_set_bytes{namespace=~"rhacs-.{20}",container!="POD",container!=""}) by (namespace, container, pod)
```

Comment (ludydoo) [reply resources/prometheus/prometheus-rules.yaml]: Fixed. I've checked if we are hitting the alerts, and we are only for 1 central instance

Comment (stehessel) [reply resources/prometheus/prometheus-rules.yaml]: But that means the alert will fire immediately? I think we need to make sure this alert is not noisy, so we need a memory buffer on all Centrals.

Comment (ludydoo) [reply resources/prometheus/prometheus-rules.yaml]: I added resources to the 1 central that would've triggered the alert

**Review by stehessel — APPROVED** [review]
Please keep an eye and this alert and tighten it if it gets noisy.

**Pull request merged**